include/openenclave/internal/atomic.h

```diff
@@ -38,4 +42,28 @@ OE_INLINE uint64_t oe_atomic_decrement(volatile uint64_t* x)
 #endif
 }
 
+OE_INLINE
+bool oe_atomic_cas(int64_t volatile* dest, int64_t old, int64_t newval)
+{
+#if defined(__GNUC__)
+    return __sync_bool_compare_and_swap(dest, old, newval);
+#elif defined(_MSC_VER)
+    return _InterlockedCompareExchange64(dest, newval, old) == old;
+#else
+#error "unsupported"
+#endif
+}
+
+OE_INLINE
+bool oe_atomic_cas_ptr(void* volatile* dest, void* old, void* newptr)
```

Comment:
Maybe to avoid type casting? or for backward compatibility with systems having 32-bit pointers? I am not sure why windows keep them separate. This is just to be safe.